chemprop/uncertainty/uncertainty_evaluator.py

```diff
@@ -0,0 +1,505 @@
+from typing import List
+
+import numpy as np
+from scipy.stats import t, spearmanr
+from scipy.special import erfinv
+
+from chemprop.data import MoleculeDataset
+from .uncertainty_calibrator import UncertaintyCalibrator
+from chemprop.train import evaluate_predictions
+
+
+class UncertaintyEvaluator:
+    """
+    A class for evaluating the effectiveness of uncertainty estimates with metrics.
+    """
+
+    def __init__(
+        self,
+        evaluation_method: str,
+        calibration_method: str,
+        uncertainty_method: str,
+        dataset_type: str,
+        loss_function: str,
+        calibrator: UncertaintyCalibrator,
+    ):
+        self.evaluation_method = evaluation_method
+        self.calibration_method = calibration_method
+        self.uncertainty_method = uncertainty_method
+        self.dataset_type = dataset_type
+        self.loss_function = loss_function
+        self.calibrator = calibrator
+
+    def raise_argument_errors(self):
+        """
+        Raise errors for incompatibilities between dataset type and uncertainty method, or similar.
+        """
+        if self.dataset_type == "spectra":
+            raise NotImplementedError(
+                "No uncertainty evaluators implemented for spectra dataset type."
+            )
+
+    def evaluate(
+        self,
+        test_data: MoleculeDataset,
+        preds: List[List[float]],
+        uncertainties: List[List[float]],
+    ):
+        """
+        Run evaluation of predicted uncertainty values
+        """
+        pass
```

Comment:
```suggestion
from abc import ABC, abstractmethod

class UncertaintyEvaluator(ABC):
    """
    A class for evaluating the effectiveness of uncertainty estimates with metrics.
    """

    def __init__(
        self,
        evaluation_method: str,
        calibration_method: str,
        uncertainty_method: str,
        dataset_type: str,
        loss_function: str,
        calibrator: UncertaintyCalibrator,
    ):
        self.evaluation_method = evaluation_method
        self.calibration_method = calibration_method
        self.uncertainty_method = uncertainty_method
        self.dataset_type = dataset_type
        self.loss_function = loss_function
        self.calibrator = calibrator

    def raise_argument_errors(self):
        """
        Raise errors for incompatibilities between dataset type and uncertainty method, or similar.
        """
        if self.dataset_type == "spectra":
            raise NotImplementedError(
                "No uncertainty evaluators implemented for spectra dataset type."
            )

    @abstractmethod
    def evaluate(
        self,
        test_data: MoleculeDataset,
        preds: List[List[float]],
        uncertainties: List[List[float]],
    ):
        """
        Run evaluation of predicted uncertainty values
        """
```

Reply:
Good suggestion with the abstract methods